examples/gno.land/r/docs/commondao/commondao_proposals.gno

```diff
@@ -0,0 +1,101 @@
+package commondao
+
+import (
+	"std"
+	"time"
+
+	"gno.land/p/nt/commondao"
+)
+
+// Proposal implements the commondao.Proposal interface
+// It contains anything useful to you in a proposal and defines how the proposal behaves
+type Proposal struct {
+	TitleProp       string
+	DescriptionProp string
+	CallerProp      std.Address
+}
+
+// Title returns the title of the proposal
+func (p *Proposal) Title() string {
+	return p.TitleProp
+}
+
+// Body returns the full description of the proposal
+func (p *Proposal) Body() string {
+	return p.DescriptionProp
+}
+
+// VotingPeriod defines the length of time the proposal is open for voting
+// Another possible implementation would be to have a constructor
+// So that each proposal can implement its own duration and tally
+func (c *Proposal) VotingPeriod() time.Duration {
+	return 24 * time.Hour // Voting period is 24 hours here
+}
+
+// Tally determines whether the proposal passes based on voting results
+// There are multiple ways to tally votes, pick the one most suited to your use case
+func (c *Proposal) Tally(r commondao.ReadonlyVotingRecord, m commondao.MemberSet) (bool, error) {
+	yesCount := r.VoteCount(commondao.ChoiceYes)
+
+	memberCount := DAO.Members().Size()
+	fivePercent := (memberCount*5 + 99) / 100 // fivePercent calculate what is 5% of the DAO's member
+
+	return yesCount >= fivePercent, nil // The proposal is considered as "passed" as soon as 5% of members voted yes
+}
+
+// Execute will execute what you decide once the proposal is considered passed
+func (p *Proposal) Execute(cur realm) error {
+	return JoinDao(cross, p.CallerProp) //When the proposal is passed the creator of the proposal joins the DAO
+}
+
+// CreateProposal constructs and submits a new proposal to the DAO
+// The proposal is stored in the ActiveProposal storage
+func CreateProposal(cur realm, title, desc string) {
+	caller := std.PreviousRealm().Address()
+
+	// Set the new proposal's title and description
+	prop := &Proposal{
+		TitleProp:       title,
+		DescriptionProp: desc,
+		CallerProp:      caller,
+	}
+
+	DAO.MustPropose(caller, prop) // Panics if the creation of the proposal fails, use Propose() if you don't want the possible panic
+}
+
+// Vote allows a DAO member to vote on an existing proposal
+// Allows member of the DAO to vote YES or NO on a proposal and give their reasoning
+func Vote(cur realm, vote bool, proposalID uint64, reason string) {
+	var choice commondao.VoteChoice
+
+	if vote {
+		choice = commondao.ChoiceYes // Other types of vote choice can be found in 'p/nt/commondao/proposal.gno'
+	} else {
+		choice = commondao.ChoiceNo
+	}
+
+	caller := std.PreviousRealm().Address()
+
+	if !DAO.Members().Has(caller) {
+		panic("must register to the DAO to vote") // Panics if the voter is not a member of the DAO
+	}
+
+	DAO.Vote(caller, proposalID, choice, reason) // logs in the user's voting choice on a specified proposal
```

Comment:
The DAO instance [will check that voter is a member](https://github.com/gnolang/gno/blob/b8a2efeef53654d97e16517a497c7f4b5f1e1512/examples/gno.land/p/nt/commondao/commondao.gno#L171) for you when the `Vote()` method is called, so you could remove the validation if you want, but make sure you use the returned error with a panic:
```suggestion
	err := DAO.Vote(caller, proposalID, choice, reason) // logs in the user's voting choice on a specified proposal
	if err != nil {
		panic(err)
	}
```